utils/custom.js

```diff
@@ -115,10 +115,17 @@ const handleAnonymousData = async (
       if (child?.data?.anon_user_info_emoji_name) {
         let childCommentLv2 = child?.latest_children?.comment || [];
         childCommentLv2 = childCommentLv2?.map((child2) => {
+          const child2_user = karmaScores.find((user) => user.user_id === child2.user_id);
           if (child2.data.anon_user_info_emoji_name) {
-            return {...child2, user_id: null, user: {}, target_feeds: []};
+            return {
+              ...child2,
+              user_id: null,
+              user: {},
+              target_feeds: [],
+              karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)
+            };
           }
-          return {...child2};
+          return {...child2, karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code seems to be handling anonymous users and regular users in a similar way, but there's a potential issue with the `user_id` field. For anonymous users, you're setting `user_id` to `null`, which is fine. However, for regular users, you're not updating the `user_id` at all, which means it could potentially contain an old or incorrect value. To ensure data consistency, consider explicitly setting the `user_id` for regular users as well.

```diff
       if (child?.data?.anon_user_info_emoji_name) {
         let childCommentLv2 = child?.latest_children?.comment || [];
         childCommentLv2 = childCommentLv2?.map((child2) => {
           const child2_user = karmaScores.find((user) => user.user_id === child2.user_id);
           if (child2.data.anon_user_info_emoji_name) {
             return {
               ...child2,
               user_id: null,
               user: {},
               target_feeds: [],
               karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)
             };
           }
-          return {...child2, karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)};
+          return {...child2, user_id: child2_user?.user_id, karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)};
         });
         return {
           ...child,
```
This change ensures that the `user_id` field is always updated correctly, whether the user is anonymous or not.


<!-- This is an auto-generated comment by OSS CodeRabbit -->